### PR TITLE
Qr payment

### PR DESCRIPTION
**Pull request by Bongz12**
QR scode payment and textbox payment

### CODE REVIEW

File: src/Components/WalletToWallet/SubViews/QRScannerPayment.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+                        style={styles.button}>
+                        <Text style={{ color: '#FFFFFF', fontSize: 12 }}>
+                            Pay with QR code
+                        </Text>
+                    </TouchableHighlight>
+                </View>
+            );
+        }
+        return (
+            <View style={{ flex: 1 }}>
```
Comment (GeromeSchutte) [inline src/Components/WalletToWallet/SubViews/QRScannerPayment.js]: please extract your styles to a stylesheet

File: src/Components/WalletToWallet/SubViews/QRScannerPayment.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        }    
+    }
+    
+    render() {
+        
+        if (!this.props.walletToWalletState.userInfo.openScanner) {
+            return (
+                <View style={styles.container}>
+                    <TextInput
+                        style={{ height: 40, borderColor: 'gray', borderWidth: 1, width:'100%' }}
```
Comment (GeromeSchutte) [inline src/Components/WalletToWallet/SubViews/QRScannerPayment.js]: please extract your styles to a stylesheet

File: src/Components/WalletToWallet/SubViews/QRScannerPayment.js [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+                        onSubmitEditing={(event) => this.handleChange(event.nativeEvent.codeStringValue)}
+                        placeholder={'pay with cellPhone Number'}
+                    />
+                    <TouchableHighlight
+                        onPress={() => this.onOpneScanner()}
+                        style={styles.button}>
+                        <Text style={{ color: '#FFFFFF', fontSize: 12 }}>
```
Comment (GeromeSchutte) [inline src/Components/WalletToWallet/SubViews/QRScannerPayment.js]: please extract your styles to a stylesheet

File: src/Components/WalletToWallet/WalletToWalletActions.js [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+export const changePaymentUserInfo = (userInfo) => {
+    return (dispatch) => {
+        dispatch(userInfonoChangedAC(userInfo));
+    };
+};
+async function onProcessPayment(username) {  
+    const usernameObj = { offset: '0',
+        limit: '50',
+        identity :  username };
+    const jwt ='Bearer eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJqYWNrc29hcCIsInVpZCI6MSwiY2giOiJZclJWSUdMbTVRIiwicm9sZXMiOiIiLCJpc3MiOiJodHRwOlwvXC9qaW5pLmd1cnUiLCJleHAiOjE1NTQ1MzkyMDZ9.nIadvzD_EsPI-toowhqfjjU6nczmXOFO5aFpUBdsw0k';
```
Comment (GeromeSchutte) [inline src/Components/WalletToWallet/WalletToWalletActions.js]: shouldn't we be not hardcoding this?

Comment (GeromeSchutte) [reply src/Components/WalletToWallet/WalletToWalletActions.js]: using the state jwt perhaps?

File: src/Components/WalletToWallet/WalletToWalletActions.js [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+    try {
+        let index=0;
```
Comment (GeromeSchutte) [inline src/Components/WalletToWallet/WalletToWalletActions.js]: this value doesnt change throughout your function. can be a const

File: src/Components/WalletToWallet/WalletToWalletReducer.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+import * as Types from './WalletToWalletActionTypes';
+
+export default (state = InitialState.walletToWalletState, action) => {
+    switch (action.type) {
+    case Types.WALLET_TO_WALLET_DETAILS:
+        return {
+            ...state,
+            ...state.walletToWalletState
+        };
+    case Types.ON_SCAN_START:
```
Comment (GeromeSchutte) [inline src/Components/WalletToWallet/WalletToWalletReducer.js]: from this line down, all cases can return 
`{
    ...state, 
    userInfo: {
        ...state.userInfo, 
        ...action.walletToWalletState.userInfo
    }
}`

File: src/Components/WalletToWallet/WalletToWalletActions.js (hunk 4 above)
Comment (Bongz12) [reply src/Components/WalletToWallet/WalletToWalletActions.js]: i cant use that one cause I will be fetched it from the QR code,thats cause the jwt in question here is that of a person being paid

File: src/Components/WalletToWallet/SubViews/QRScannerPayment.js [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+        this.props.WalletToWalletActions.qrScan(event);
+    }
+    handleChange (event){
+        this.props.WalletToWalletActions.qrScan(event);
+    }
+    async requestCameraPermission() {
+        try {
+            const granted = await PermissionsAndroid.request(
+                PermissionsAndroid.PERMISSIONS.CAMERA,{
+                    'title': 'CameraExample App Camera Permission',
```
Comment (GeromeSchutte) [inline src/Components/WalletToWallet/SubViews/QRScannerPayment.js]: We shouldn't still be using the example text

Comment (Bongz12) [reply src/Components/WalletToWallet/SubViews/QRScannerPayment.js]: thats now fixed

File: src/Components/WalletToWallet/WalletToWalletActions.js [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+        limit: '50',
+        identity :  username };
+    const jwtValue =jwt;
+    try {
+        const index=0; 
+        const result= await API.getUserByUsername(usernameObj,jwtValue);
+    
+        return { 
+            userId: result.data[index].userId
+            ,cellNo: result.data[index].phone1
```
Comment (GeromeSchutte) [inline src/Components/WalletToWallet/WalletToWalletActions.js]: how come your commas are on the start of the line?

Comment (Bongz12) [reply src/Components/WalletToWallet/WalletToWalletActions.js]: just changed this

**Pull request merged**